src/compile/data/fold.ts

```diff
@@ -0,0 +1,53 @@
+import {FoldTransform as VgFoldTransform} from 'vega-typings';
+import {FoldTransform} from '../../transform';
+import {duplicate} from '../../util';
+import {DataFlowNode} from './dataflow';
+
+/**
+ * A class for flatten transform nodes
+ */
+export class FoldTransformNode extends DataFlowNode {
+
+  public clone() {
+    return new FoldTransformNode(this.parent, duplicate(this.transform));
+  }
+
+  constructor(parent: DataFlowNode, private transform: FoldTransform) {
+    super(parent);
+    this.transform.as = this.getNames();
+  }
+
+  public producedFields() {
+    const names = this.getNames();
```

Comment:
It's weird that you're calling `getNames()` again in producedFields(). 

The whole point of setting as in constructor is that you can just simply uses `as` to determine producedFields.

Reply:
Let me fix it :)